internal/character/blade/ult.go

```diff
@@ -0,0 +1,64 @@
+package blade
+
+import (
+	"math"
+
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Ult key.Attack = "blade-ult"
+)
+
+func (c *char) Ult(target key.TargetID, state info.ActionState) {
+	c.engine.SetHP(info.ModifyAttribute{
+		Key:    key.Reason(Ult),
+		Target: c.id,
+		Source: c.id,
+		Amount: c.engine.Stats(c.id).MaxHP() * 0.5,
+	})
+
+	hpTally := math.Min(c.hpLoss, 0.9*c.engine.Stats(c.id).MaxHP())
+
+	// TODO: Seperate Tally?
+	e1TallyMod := 0.0
+
+	if c.info.Eidolon >= 1 {
+		e1TallyMod = 1.5
+	}
+
+	// Primary Target
+	c.engine.Attack(info.Attack{
+		Key:        Ult,
+		Source:     c.id,
+		Targets:    []key.TargetID{target},
+		DamageType: model.DamageType_WIND,
+		AttackType: model.AttackType_ULT,
+		BaseDamage: info.DamageMap{
+			model.DamageFormula_BY_ATK:    ultSingleAtk[c.info.UltLevelIndex()],
+			model.DamageFormula_BY_MAX_HP: ultSingleHP[c.info.UltLevelIndex()],
+		},
+		DamageValue:  hpTally * (ultSingleTally[c.info.UltLevelIndex()] + e1TallyMod),
+		StanceDamage: 60.0,
+		EnergyGain:   5.0,
+	})
+
+	// Adjacent Targets
+	c.engine.Attack(info.Attack{
+		Key:        Ult,
```

Comment:
nit: probably want separate key for primary and adjacent (I created separate keys for arlan ult since different scalings)